torch/nn/modules/dropout.py

```diff
@@ -184,6 +184,46 @@ def forward(self, input: Tensor) -> Tensor:
         return F.alpha_dropout(input, self.p, self.training)
 
 
+class FeatureDropout(_DropoutNd):
+    r"""Randomly masks out entire channels (a channel is a feature map,
+    e.g. the :math:`j`-th channel of the :math:`i`-th sample in the batch input
+    is a tensor :math:`\text{input}[i, j]`) of the input tensor).
+
+    Each channel will be zeroed out independently on every forward call with
+    probability :attr:`p` using samples from a Bernoulli distribution.
+
+    Internally used by `DropoutXd` making stack traces less confusing.
+
+    As described in the paper
+    `Efficient Object Localization Using Convolutional Networks`_ ,
+    if adjacent pixels within feature maps are strongly correlated
+    (as is normally the case in early convolution layers) then i.i.d. dropout
+    will not regularize the activations and will otherwise just result
+    in an effective learning rate decrease.
+
+    Args:
+        p (float, optional): probability of an element to be zeroed. Default: 0.5
+        inplace (bool, optional): If set to ``True``, will do this operation
+            in-place
+
+    Shape:
+        - Input: :math:`(N, C, D, H, W)`
```

Comment:
What is semantics of D? Would FeatureDropout also work for 1-d case, i.e. NCT? 2-d case, i.e. NCHW?

Reply:
I followed `FeatureAlphaDropout` implementation which seems to differ only in [additionally applying additional factor to the noise](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/aten/src/ATen/native/Dropout.cpp#L56) and therefore I took shapes from there.

AFAIU `FeatureDropout` [just requires >=2 dimensionality](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/aten/src/ATen/native/Dropout.cpp#L14). Should I change it and also correct [existing FeatureAlphaDropout docs then](https://github.com/pytorch/pytorch/blob/df837d0384398f1fac1000189d62b05e7ff0240d/torch/nn/modules/dropout.py#L218)?